src/grid.jl

```diff
@@ -0,0 +1,100 @@
+module Grid
+
+import ..Renderables: AbstractRenderable
+import ..Measures: Measure, default_size
+import ..Layout: PlaceHolder, vstack
+import ..Compositors: Compositor
+
+import Term: calc_nrows_ncols
+
+export grid
+
+include("_compositor.jl")
+
+# ---------------------------------------------------------------------------- #
+#                                     GRID                                     #
+# ---------------------------------------------------------------------------- #
+
+"""
+    grid(
+        rens::Union{Nothing,AbstractVector{<:AbstractRenderable}} = nothing;
+        placeholder::Union{Nothing,AbstractRenderable} = nothing,
+        aspect::Union{Nothing,Number,NTuple} = nothing,
+        layout::Union{Nothing,Tuple} = nothing, 
+        pad::Union{Tuple,Integer} = 0,
+    )
+
+Construct a grid from a `AbstractVector` of `AbstractRenderable`s.
+
+Lays out the renderables to createa a grid with the desired aspect ratio or
+layout (number of rows, number of columns). If no renderables are passed it
+creates placeholders.
+"""
+function grid(
+    rens::Union{Nothing,AbstractVector{<:AbstractRenderable}} = nothing;
+    placeholder::Union{Nothing,AbstractRenderable} = nothing,
+    aspect::Union{Nothing,Number,NTuple} = nothing,
+    layout::Union{Nothing,Tuple,Expr} = nothing,
+    show_placeholder::Bool = false,
+    pad::Union{Tuple,Integer} = 0,
+)
+    (isnothing(layout) && isnothing(rens)) && error("Grid: layout or aspect required")
+
+    if !isnothing(rens) && layout isa Expr
+        n, kw = 0, Dict()
+        sz = size(Measure(first(rens).segments))
+        for e in get_elements_and_sizes(layout; placeholder_size = sz)
+            (nm = e.args[1]) === :_ && continue
+            kw[nm] = rens[n += 1]
+        end
+        return Compositor(layout; placeholder_size = sz, check = false, pairs(kw)...)
```

Comment:
This should be `Renderable(Compositor(...))`  to keep it consistent with the other behavior where `grid` returns a renderable

Reply:
Thanks, fixed.